packages/bigquery/src/dataset.js

```diff
@@ -414,4 +470,15 @@ Dataset.prototype.table = function(id) {
  */
 common.paginator.extend(Dataset, ['getTables']);
 
+/*! Developer Documentation
+ *
+ * All async methods (except for streams) will return a Promise in the event
+ * that a callback is omitted.
+ */
+common.util.promisify(Dataset, {
```

Comment:
What do you think about having `Promisify` as a base class that `Service` inherits from? Not sure if it's doable without micro-configuration.